File: leetcode/226-invert-binary-tree/solution.go
func invertTree(root *TreeNode) *TreeNode {
  if (root == nil) {
    return root;
  }

  var t *TreeNode = root.Left;
  root.Left = root.Right;
  root.Right = t;

  invertTree(root.Left);
  invertTree(root.Right);

  return root;
}

File: leetcode/226-invert-binary-tree/solution.java
class Solution {
    public TreeNode invertTree(TreeNode root) {
        if (root == null) return null;
        final var n = root.left;
        root.left = root.right;
        root.right = n;

        invertTree(root.left);
        invertTree(root.right);

        return root;
    }
}
